system_tests/run_emulator.py

```diff
@@ -95,16 +101,36 @@ def datastore_wait_ready(popen):
         emulator_ready = popen.stderr.readline() == _DS_READY_LINE
 
 
-def pubsub_wait_ready(popen):
-    """Wait until the pubsub emulator is ready to use.
+def wait_ready_prefix(popen, prefix):
+    """Wait until the a process encounters a line with matching prefix.
 
     :type popen: :class:`subprocess.Popen`
     :param popen: An open subprocess to interact with.
+
+    :type prefix: str
+    :param prefix: The prefix to match
     """
     emulator_ready = False
     while not emulator_ready:
-        emulator_ready = popen.stderr.readline().startswith(
-            _PS_READY_LINE_PREFIX)
+        emulator_ready = popen.stderr.readline().startswith(prefix)
+
+
+def pubsub_wait_ready(popen):
+    """Wait until the Pub / Sub emulator is ready to use.
+
+    :type popen: :class:`subprocess.Popen`
+    :param popen: An open subprocess to interact with.
+    """
+    wait_ready_prefix(popen, _PS_READY_LINE_PREFIX)
+
+
+def bigtable_wait_ready(popen):
```

Comment:
Would it make sense to make sense to just pass `_*_PREFIX`s to `wait_ready` and then just remove these functions?